From 9f4e0d9d611ee8580eedb61fce4af8686f926a9c Mon Sep 17 00:00:00 2001
From: Jiaxing Li <jiaxing.li@oss.qualcomm.com>
Date: Thu, 9 Oct 2025 16:47:54 +0800
Subject: [PATCH] minkipc:compilation: Fix compilation issue

- To fix compilation error: #warning "<cstdbool> is deprecated in C++17, remove the #include" [-Werror=cpp].
- Disable downloading minkidlc in CMakeLists.txt.

Signed-off-by: Jiaxing Li <jiaxing.li@oss.qualcomm.com>
Upstream-Status: Inappropriate [meta-qcom specific]
---
 CMakeLists.txt | 15 +++++++++------
 1 file changed, 9 insertions(+), 6 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 346e1bc..7f12f3f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -17,19 +17,22 @@ option(BUILD_GPFS_LISTENER "Build GP File system Listener" TRUE)

 include(GNUInstallDirs)

-set(MINKIDLC_BIN_DIR ${CMAKE_CURRENT_SOURCE_DIR}/minkidlc)
+set(MINKIDLC_BIN_DIR "${CMAKE_CURRENT_SOURCE_DIR}/minkidlc" CACHE PATH "Directory containing the minkidlc executable")

-file(DOWNLOAD
-  https://github.com/quic/mink-idl-compiler/releases/download/v0.2.0/idlc
-  ${MINKIDLC_BIN_DIR}/minkidlc
-)
+if ("${MINKIDLC_BIN_DIR}" STREQUAL "${CMAKE_CURRENT_SOURCE_DIR}/minkidlc")
+	file(DOWNLOAD
+	https://github.com/quic/mink-idl-compiler/releases/download/v0.2.0/idlc
+	${MINKIDLC_BIN_DIR}/minkidlc
+	)

-file(CHMOD ${MINKIDLC_BIN_DIR}/minkidlc PERMISSIONS OWNER_EXECUTE OWNER_WRITE OWNER_READ)
+	file(CHMOD ${MINKIDLC_BIN_DIR}/minkidlc PERMISSIONS OWNER_EXECUTE OWNER_WRITE OWNER_READ)
+endif()

 add_compile_options(
	-Wall -Wextra -Werror -Wshadow -Wcast-align
	-Wmissing-declarations -Wformat-security -Wmissing-noreturn
	-Wdeprecated -fPIC
+	-Wno-error=cpp
 )

 add_subdirectory(libminkadaptor)
--
2.34.1

